Add web framework

andrew-js-wright opened the pull request:
Wanted to get this in early as it is the boiler plate to get a functioning web app. The Postgres integration is still required but I wanted to get the schema in place before doing that.

andrew-js-wright committed:
26: Add postgres database to developer setup

andrew-js-wright committed:
Add basic node application with gov styles

andrew-js-wright committed:
Merge branch 'master' into feature/wmt-26/om-caseload-utilisation

andrew-js-wright committed:
Move  to

andrew-js-wright committed:
Add gitignore

andrew-js-wright committed:
Add a runable express application with govuk styles

tswann commented on wmt-web/README.md:
```diff
@@ -0,0 +1,16 @@
+It is a node.js application using the express web framework.
+
+## Prerequisites
+- Node v7.6.x
```
Might be worth checking this with Steven, but I have a feeling we were on the Node 6.9.0 (LTS) for the production APVS app, which would be recommendation here also.

We also should manage Node versions via NVM.

andrew-js-wright replied:
Good point. I've checked on nvm and the most up-to-date LTS appears to be 6.10.0, happy if we use that from here on out?

tswann reviewed (approved):
Looks good - one comment re. Node version

andrew-js-wright committed:
Updated node version in README to 6.10.0 (Latest LTS)

Also move migrations to the parent directory because it not only affects the web
package but the whole application

andrew-js-wright committed:
Add database migrations and seeding

Use knex's build in migration and seed functionality in order to setup the
database.

New migrations and seed files can be created using the knex cli.

Fix the docker clean up script not removing volumes from the intermediary
virtual machine.

andrew-js-wright commented on config.js:
```diff
@@ -1,6 +1,6 @@
 module.exports = {
   // DB
-  DATABASE_SERVER: 'localhost',
+  DATABASE_SERVER: '192.168.99.100',
```
This is the IP of the docker virtual host. I tried to make it a little tidier by using the command `docker-machine ip default` but it was complaining because it didn't have the shell's context. I could probably fix this with a little more time but it appears to be static on my machine anyway. Would be interested to know if this works for you. If not I'll have another bash at it.

andrew-js-wright committed:
Move database migrations and data into `wmt-data` subproject
